### PR TITLE
Final change the protobufs (supporting v4 and v5 schemas)

### PR DESCRIPTION
**Pull request by placer14**
Many changes are in this PR. They stem from the protobufs being updated and propagate out to the rest of the app and unit tests (which build and pass). QA tests are currently untouched and still need attention. This also doesn't support any of the untested examples (legacy support scenarios, primarily).

This is a good milestone, though.

### CODE REVIEW

**Comment by drwasho** [comment]
My feedback:

- [ ] 1. We have a bug where the API fails when trying to spend ETH/TETH, but the transactions still go through https://github.com/OpenBazaar/openbazaar-go/issues/1730. Does it make sense to solve it in this PR, or do we want to merge this and fix the bug as a separate PR?
 
- [ ] 2. Updating the websocket message to be consistent with the REST API is something that I think this PR should address: https://github.com/OpenBazaar/openbazaar-go/issues/1696

- [ ] 3. Related to (2) is the bug reported here: https://github.com/OpenBazaar/openbazaar-go/issues/1701

- [ ] 4. In the interests of time we should merge this PR and follow-up with updated QA tests as soon as possible. To expedite that, @placer14 it would be really helpful if you could list the QA tests that are failing and require refactoring.

File: pb/protos/contracts.proto [hunk 1]
```diff
@@ -222,13 +215,15 @@ message Order {
...
-    string paymentAddress                     = 3;
-    //uint64 requestedAmount             = 4 [json_name = ""];
-    repeated RatingSignature ratingSignatures = 5;
-    CurrencyValue requestedAmountValue             = 6 [json_name = "requestedAmount"];
+    string paymentAddress                      = 3;
+    uint64 requestedAmount                     = 4 [deprecated = true];
+
+    repeated RatingSignature ratingSignatures  = 5;
+    string bigRequestedAmount                  = 6;
+    CurrencyDefinition requestedAmountCurrency = 7;
```
Comment (cpacia) [inline pb/protos/contracts.proto]: Does look like we're using this anywhere. Not sure it needs to be included.

File: core/listings.go [hunk 2]
```diff
@@ -310,13 +314,14 @@ func (n *OpenBazaarNode) extractListingData(listing *pb.SignedListing) (repo.Lis
 		ContractType:       listing.Listing.Metadata.ContractType.String(),
 		Description:        listing.Listing.Item.Description[:descriptionLength],
 		Thumbnail:          repo.ListingThumbnail{listing.Listing.Item.Images[0].Tiny, listing.Listing.Item.Images[0].Small, listing.Listing.Item.Images[0].Medium},
-		Price:              repo.CurrencyValue{Currency: defn, Amount: amt},
+		Price:              priceValue,
 		Modifier:           listing.Listing.Metadata.PriceModifier,
```
Comment (cpacia) [inline core/listings.go]: using deprecated metadata.PriceModifier rather than item.PriceModifier

**Comment by cpacia** [comment]
It appears `func (l *Listing) GetPriceModifier() (float32, error) ` is returning from the deprecated metadata.PriceModifier rather than new item.PriceModifier

**Comment by cpacia** [comment]
I have opened an new issue for the final changes that we are going to need. https://github.com/OpenBazaar/openbazaar-go/issues/1738

So far this PR LGTM other than the few comments I made.

**Comment by drwasho** [comment]
- [ ] There is inconsistency when fetching a listing:

`GET /ob/listing/:slug`

image

`GET /ob/listing/ipfs/:hash`

image

**Comment by drwasho** [comment]
- [ ] Estimating a purchase (`POST /ob/estimatetotal`) && making a purchase order (`POST /ob/purchase`) is failing with the following payload and response:

Payload:

```JSON
{
    "shipTo": "Elwood Blues",
    "address": "1060 W Addison",
    "city": "Chicago",
    "state": "Illinois",
    "countryCode": "UNITED_STATES",
    "postalCode": "60613",
    "addressNotes": "",
    "items": [
        {
            "listingHash": "QmTeGATiiTEbNwkLosSRivSApstXLof45jWRjSAVoB1QUv",
            "bigQuantity": "1",
            "options": [
                {
                    "name": "Sizes",
                    "value": "Small"
                },
                {
                    "name": "Color",
                    "value": "Red"
                }
            ],
            "shipping": {
                "name": "Worldwide",
                "service": "Standard"
            },
            "memo": "thanks!",
            "coupons": []
        }
    ],
    "moderator": "QmcdkKM2fWCKzTZdpjEY6abzRbDhRRJvNmqwNKVyiDtGky",
    "paymentCoin": "TLTC"
}
```

Reponse:

```JSON
{
    "success": false,
    "reason": "unknown currency"
}
```

**Pull request merged**